### PR TITLE
add GetDBByNameAndHostInfo() and GetDBsByHostInfo()

### PR DESCRIPTION
**Pull request by gengxiaoyun**
add GetDBByNameAndHostInfo() and GetDBsByHostInfo()

### CODE REVIEW

File: internal/app/metadata/db_repository.go [hunk 1]
```diff
@@ -178,6 +178,75 @@ func (dr *DBRepo) GetDBByNameAndClusterInfo(name string, clusterID, clusterType
...
 
+// GetDBByNameAndHostInfo gets a database by the db name and host info from the middleware
+func (dr *DBRepo) GetDBByNameAndHostInfo(name, hostIP string, portNum int) (metadata.DB, error) {
+	sql := `
+		select db.id, db.db_name, db.cluster_id, db.cluster_type, db.env_id, db.del_flag, db.create_time, db.last_update_time
+		from t_meta_db_info as db
+		inner join t_meta_mysql_server_info as ms
+		on db.cluster_id = ms.cluster_id 
+		where db.del_flag = 0
+		and db.db_name = ?
```
Comment (romberli) [inline internal/app/metadata/db_repository.go:189]: missed and ms.del_flag = 0

File: internal/app/metadata/db_repository.go [hunk 2]
```diff
@@ -178,6 +178,75 @@ func (dr *DBRepo) GetDBByNameAndClusterInfo(name string, clusterID, clusterType
 	}
 }
 
+// GetDBByNameAndHostInfo gets a database by the db name and host info from the middleware
+func (dr *DBRepo) GetDBByNameAndHostInfo(name, hostIP string, portNum int) (metadata.DB, error) {
+	sql := `
+		select db.id, db.db_name, db.cluster_id, db.cluster_type, db.env_id, db.del_flag, db.create_time, db.last_update_time
+		from t_meta_db_info as db
+		inner join t_meta_mysql_server_info as ms
+		on db.cluster_id = ms.cluster_id
```
Comment (romberli) [inline internal/app/metadata/db_repository.go:187]: missed  and db.cluster_type = 1

File: internal/app/metadata/db_repository.go [hunk 3]
```diff
@@ -178,6 +178,75 @@ func (dr *DBRepo) GetDBByNameAndClusterInfo(name string, clusterID, clusterType
...
+
+// GetDBsByHostInfo gets databases by the host info from the middleware
+func (dr *DBRepo) GetDBsByHostInfo(hostIP string, portNum int) ([]metadata.DB, error) {
+	sql := `
+		select db.id, db.db_name, db.cluster_id, db.cluster_type, db.env_id, db.del_flag, db.create_time, db.last_update_time
+		from t_meta_db_info as db
+		inner join t_meta_mysql_server_info as ms
+		on db.cluster_id = ms.cluster_id 
+		where db.del_flag = 0
+		and ms.host_ip = ?
```
Comment (romberli) [inline internal/app/metadata/db_repository.go:223]: missed and ms.del_flag = 0

File: internal/app/metadata/db_repository.go [hunk 4]
```diff
@@ -178,6 +178,75 @@ func (dr *DBRepo) GetDBByNameAndClusterInfo(name string, clusterID, clusterType
...
+		and ms.port_num = ?;
+	`
+	log.Debugf("metadata DBRepo.GetDBsByHostInfo() sql: \n%s\nplaceholders: %s, %d", sql, hostIP, portNum)
+	result, err := dr.Execute(sql, hostIP, portNum)
+	if err != nil {
+		return nil, err
+	}
+	// init []*DBInfo
+	dbInfoList := make([]*DBInfo, result.RowNumber())
```
Comment (romberli) [inline internal/app/metadata/db_repository.go:232]: make a []metadata.DB directly, doesn't need to make a []*DBInfo temporarily

File: internal/app/metadata/db_repository.go [hunk 5]
```diff
@@ -178,6 +178,75 @@ func (dr *DBRepo) GetDBByNameAndClusterInfo(name string, clusterID, clusterType
...
+	for i := range dbInfoList {
+		dbInfoList[i] = NewEmptyDBInfoWithGlobal()
+	}
+	// map to struct
+	err = result.MapToStructSlice(dbInfoList, constant.DefaultMiddlewareTag)
+	if err != nil {
+		return nil, err
+	}
+	// init []metadata.DB
+	dbList := make([]metadata.DB, result.RowNumber())
```
Comment (romberli) [inline internal/app/metadata/db_repository.go:242]: as making []metadata.DB directly, this line and following 3 lines doesn't need anymore

**Review by romberli — CHANGES_REQUESTED** [review]

File: internal/app/metadata/db_repository.go (hunk 1 above)
Comment (gengxiaoyun) [reply internal/app/metadata/db_repository.go:183]: done

File: internal/app/metadata/db_repository.go (hunk 2 above)
Comment (gengxiaoyun) [reply internal/app/metadata/db_repository.go:179]: done

File: internal/app/metadata/db_repository.go [hunk 6]
```diff
@@ -178,6 +178,75 @@ func (dr *DBRepo) GetDBByNameAndClusterInfo(name string, clusterID, clusterType
...
+	}
+}
+
+// GetDBsByHostInfo gets databases by the host info from the middleware
+func (dr *DBRepo) GetDBsByHostInfo(hostIP string, portNum int) ([]metadata.DB, error) {
+	sql := `
+		select db.id, db.db_name, db.cluster_id, db.cluster_type, db.env_id, db.del_flag, db.create_time, db.last_update_time
+		from t_meta_db_info as db
+		inner join t_meta_mysql_server_info as ms
+		on db.cluster_id = ms.cluster_id
```
[the remark this answers is not in the document]
Comment (gengxiaoyun) [reply internal/app/metadata/db_repository.go:215]: done

File: internal/app/metadata/db_repository.go (hunk 3 above)
Comment (gengxiaoyun) [reply internal/app/metadata/db_repository.go:219]: done

File: internal/app/metadata/db_repository.go (hunk 4 above)
Comment (gengxiaoyun) [reply internal/app/metadata/db_repository.go]: done

File: internal/app/metadata/db_repository.go (hunk 5 above)
Comment (gengxiaoyun) [reply internal/app/metadata/db_repository.go]: done

**Review by romberli — APPROVED** [review]

**Pull request merged**